### PR TITLE
Added support for adding an infohash directly to the Add Link option

### PR DESCRIPTION
**Pull request by PiwwowPants**
* A user can now type the infohash of a torrent directly into "Add Link" and it will add the torrent
     * Any string not containing the MAGNET_PREFIX or HTTP(S)_PREFIX will automatically attempt to convert to an infohash using the new INFOHASH_PREFIX

### CODE REVIEW

File: app/src/main/java/org/proninyaroslav/libretorrent/core/utils/Utils.java [hunk 1]
```diff
@@ -68,6 +68,7 @@
     public static final String MAGNET_PREFIX = "magnet";
     public static final String HTTP_PREFIX = "http";
     public static final String HTTPS_PREFIX = "https";
+    public static final String INFOHASH_PREFIX = "magnet:?xt=urn:btih:";
```
Comment (PiwwowPants) [inline app/src/main/java/org/proninyaroslav/libretorrent/core/utils/Utils.java]: Is there a reason to not include "btih:" on this string? I assume this app only uses the BitTorrent protocol but if it does not, we'd need to remove this and require the user to add that part of the prefix themselves / add a spinner for options to select.

**Comment by proninyaroslav** [comment]
I think you should do the same as in `qbittorrent` (incl. validation, because we can't know whether the input data is correct):

**Comment by PiwwowPants** [comment]
I agree with the validation but should we leave http and https as the fall back then? I didn't really validate the infohashes being real since I put validation on the HTTP URLs, so anything you type will try to be an infohash and if it's fake it just won't work. 

I guess I could just throw a generic error in the else statement and call it a day.

**Comment by proninyaroslav** [comment]
I think you are right, let it remain as it is. In either case, the parser will catch any errors.

**Comment by PiwwowPants** [comment]
Sounds good. I was playing with it after you mentioned it and without an upsetting rewrite, it would require me to run the value through the regex-based matcher twice which didn't seem great.
If someday additional values are added, that bridge can be crossed.

**Pull request merged**